data/index.js

```diff
@@ -80,18 +80,25 @@ function addAssignment(assignmentName, prompt, dueDate) {
 	if (typeof dueDate != "string") {
 		return Promise.reject("Due date must be provided");
 	}
+	return courses().then(collection => {
+		return collection.findOne({_id: classID}).then((course) => {
+			return course.studentIDs;
+		});
+	}).then((ids) => {
+		let submissions = ids.map((x) => {return {studentID: x, grade: 'NA'}});
+        let newAssignment = {
+            _id: uuid(),
+            assignmentName: assignmentName,
+            prompt: prompt,
+            dueDate: dueDate,
+            submissions: submissions
+        };
 
```

Comment:
My Javascript ain't the best, but isn't it impossible to reach this part of the code because you already returned a promise?